scipy/stats/_continuous_distns.py

```diff
@@ -3639,8 +3706,351 @@ def alphanot1func(alpha, beta, TH, aTH, bTH, cosTH, tanTH, W):
     def _argcheck(self, alpha, beta):
         return (alpha > 0) & (alpha <= 2) & (beta <= 1) & (beta >= -1)
 
+    @staticmethod
+    def _cf(t, alpha, beta):
+        Phi = lambda alpha, t: np.tan(np.pi*alpha/2) if alpha != 1 else -2.0*np.log(np.abs(t))/np.pi
+        return np.exp(-(np.abs(t)**alpha)*(1-1j*beta*np.sign(t)*Phi(alpha, t)))
+        
+    @staticmethod
+    def _pdf_from_cf_with_fft(cf, h=0.01, q=9):
+        """Calculates pdf from cf using fft. Using region around 0 with N=2**q points
+        separated by distance h. As suggested by [MS].
+        """
+        N = 2**q
+        n = np.arange(1,N+1)
+        density = ((-1)**(n-1-N/2))*np.fft.fft(((-1)**(n-1))*cf(2*np.pi*(n-1-N/2)/h/N))/h/N
+        x = (n-1-N/2)*h
+        return (x, density)
+    
+    @staticmethod
+    def _pdf_single_value_best(x, alpha, beta):
+        if alpha != 1. or (alpha == 1. and beta == 0.):
+            return levy_stable_gen._pdf_single_value_zolotarev(x, alpha, beta)
+        else:
+            return levy_stable_gen._pdf_single_value_cf_integrate(x, alpha, beta)
+    
+    @staticmethod
+    def _pdf_single_value_cf_integrate(x, alpha, beta):
+        cf = lambda t: levy_stable_gen._cf(t, alpha, beta)
+        return integrate.quad(lambda t: np.real(np.exp(-1j*t*x)*cf(t)), -np.inf, np.inf, limit=1000)[0]/np.pi/2
+    
+    @staticmethod
+    def _pdf_single_value_zolotarev(x, alpha, beta):
+        """Calculate pdf using Zolotarev's methods as detailed in [BS].
+        """
+        zeta = -beta*np.tan(np.pi*alpha/2.)
+        if alpha != 1:
+            x0 = x + zeta  # convert to S_0 parameterization
+            xi = np.arctan(-zeta)/alpha
+            
+            def V(theta):
+                return np.cos(alpha*xi)**(1/(alpha-1)) * \
+                                (np.cos(theta)/np.sin(alpha*(xi+theta)))**(alpha/(alpha-1)) * \
+                                (np.cos(alpha*xi+(alpha-1)*theta)/np.cos(theta))
+            if x0 > zeta:
+                def g(theta):
+                    return V(theta)*np.real(np.complex(x0-zeta)**(alpha/(alpha-1)))
+                
+                def f(theta):
+                    return g(theta) * np.exp(-g(theta))
+
+                # spare calculating integral on null set
+                # use isclose as macos has fp differences
+                if np.isclose(-xi, np.pi/2):
```

Comment:
`isclose` has a default relative tolerance of 1e-5. I think what you intended here is to check for equality with close to machine precision? this throws away too much accuracy it looks like.